src/corelibs/U2View/src/ov_phyltree/TreeViewer.cpp

```diff
@@ -650,28 +652,55 @@ void TreeViewerUI::updateSettings() {
         scene()->update();
     }
 }
-void TreeViewerUI::updateTextSettings() {
-    QList<QGraphicsItem *> updatingItems = scene()->selectedItems();
-    QColor curColor = qvariant_cast<QColor>(getOptionValue(LABEL_COLOR));
-    if (updatingItems.isEmpty()) {
-        updatingItems = items();
-
-        QList<QGraphicsItem *> legendChildItems = legend->childItems();
-        if (!legendChildItems.isEmpty()) {
-            QGraphicsSimpleTextItem *legendText = dynamic_cast<QGraphicsSimpleTextItem *>(legendChildItems.first());
-            if (legendText) {
-                legendText->setBrush(curColor);
-            }
+void TreeViewerUI::getSelectedBranchSettings(QColor &color, QFont &font) const {
+    QList<QGraphicsItem *> selectedItems = scene()->selectedItems();
+    if (selectedItems.isEmpty()) {
+        return;
+    }
+    int maxLength = 0;
+    GraphicsBranchItem *rootItem = nullptr;
+    for (QGraphicsItem *graphItem : qAsConst(selectedItems)) {
+        auto *branchItem = dynamic_cast<GraphicsBranchItem *>(graphItem);
+        CHECK(branchItem != nullptr, )
+        if (branchItem->getBranchLength() >= maxLength) {
+            maxLength = branchItem->getBranchLength();
+            rootItem = branchItem;
         }
     }
+    CHECK(rootItem != nullptr, )
+    if (rootItem->getDistanceText() != nullptr) {
+        font = rootItem->getDistanceText()->font();
+        color = rootItem->getDistanceText()->brush().color();
+    }
+}
 
-    QFont curFont = qvariant_cast<QFont>(getOptionValue(LABEL_FONT));
+void TreeViewerUI::updateTextSettings(TreeViewOption option) {
+    QList<QGraphicsItem *> updatingItems = scene()->selectedItems().isEmpty() ? scene()->items() : scene()->selectedItems();
```

Comment:
Please, don't use such a long and difficult conditions. Usual `if-else` is more convenient in this case.